src/main_thread/decrypt/utils/clean_old_loaded_sessions.ts

```diff
@@ -30,7 +30,7 @@ export default async function cleanOldLoadedSessions(
   loadedSessionsStore: LoadedSessionsStore,
   limit: number,
 ): Promise<void> {
-  if (limit < 0 || limit >= loadedSessionsStore.getLength()) {
+  if (limit < 0 || limit > loadedSessionsStore.getLength()) {
```

Comment:
I'm not sure this is right:
If `maxSessionCacheSize` was set to `2` in the API, then `limit` here is equal to `maxSessionCacheSize - 1` so limit is `1`.

So before creating first session `loadedSessionsStore.getLength()` is `1`.
Then `1 > 1` returns `false` and we clean old session,
thus we should be able to have 2 sessions!

Reply:
It seems to me that you're right though we wouldn't be closing anything thanks to:

>   const toDelete = entries.length - limit;

I'm not sure of what I was thinking here. 

Can you fix it on your side? It's hard to develop in my current setup.

Reply:
Ok, I have changed it, I removed the change on this file.
Basically this file was already fine, and behaves like `cleanOldStoredPersistentInfo`.

The only change needed is `create_or_load_session.ts`, that have the logic of setting the limit minus 1 because it's about to create a session. 

The equivalent is already done for persistent sessions in the content decryptor:

```js
cleanOldStoredPersistentInfo(
  persistentSessionsStore,
  EME_MAX_STORED_PERSISTENT_SESSION_INFORMATION - 1,
);
```